packages/mattermost-redux/src/reducers/entities/channels.ts

```diff
@@ -480,25 +501,29 @@ export function myMembers(state: RelationOneToOne<Channel, ChannelMembership> =
 }
 
 function receiveChannelMember(state: RelationOneToOne<Channel, ChannelMembership>, received: ChannelMembership) {
-    const member = state[received.channel_id];
+    const existingChannelMember = state[received.channel_id];
+    let updatedChannelMember = {...received};
+
+    if (isEqual(existingChannelMember, updatedChannelMember)) {
```

Comment:
Maybe move this check to be after the below if statement? That way, we'll also avoid updating state if the the updated last_viewed_at is less than the existing last_viewed_at

Reply:
We chatted and this is actually correct as-is. The only case we'd care about checking this afterwards is if the below if statement could make those two objects equal, but that's impossible since the if statement requires that the `last_viewed_at` values were different (because it's a `<` instead of a `<=`)